### PR TITLE
Added JSON-RPC and CLI 'payto_slp' command to send tokens, also CLI command 'create_slp'

### PR DESCRIPTION
**Pull request by cculianu**
@pokkst asked for this and he may be testing it.

- Works like the already-existing command `payto`, except it only deals in tokens. All BCH and change, etc, goes back to wallet. (So it only generates pure SLP tx's).
- It enforces SLP address for the destination payee.
- Code is taken mostly from main_window.py's tx construction code in the send tab.

**I could use your review on this, @jcramer and @markblundeberg !**

Here is the help for it (from this the JSON-RPC method can easily be deduced):

```
usage: electron-cash payto_slp [-h] [-v] [-D ELECTRON_CASH_PATH] [-P]
                              [-w WALLET_PATH] [-wp WALLET_PASSWORD]
                              [--testnet] [-f FEE] [-F FROM_ADDR]
                              [-c CHANGE_ADDR] [-u] [-W PASSWORD]
                              [--locktime LOCKTIME]
                              token_id destination_slp amount_slp

Create an SLP token transaction.

positional arguments:
  token_id              SLP token id (64 character hex string)
  destination_slp       SLP address; where to send the token
  amount_slp            Amount to be sent (in token units, floats ok)

optional arguments:
  -h, --help            show this help message and exit
  -f FEE, --fee FEE     Transaction fee (in BCH)
  -F FROM_ADDR, --from_addr FROM_ADDR
                        Source address (must be a wallet address; use sweep to
                        spend from non-wallet address).
  -c CHANGE_ADDR, --change_addr CHANGE_ADDR
                        Change address. Default is a spare address, or the
                        source address if it's not in the wallet
  -u, --unsigned        Do not sign transaction
  -W PASSWORD, --password PASSWORD
                        Password
  --locktime LOCKTIME   Set locktime block number
```

---

Also in this PR, `create_slp`:

- `create_slp` is identical to `create` (CLI-only, no JSON-RPC impl), except it creates a wallet of type `slp_standard` as opposted to `create`'s impl. which creates `standard` wallets.

### CODE REVIEW

File: lib/commands.py [hunk 1]
```diff
@@ -495,6 +497,70 @@ def paytomany(self, outputs, fee=None, from_addr=None, change_addr=None, nocheck
...
+                    change_addr = coins[0]['address']
+            bch_outputs.append((change_addr.kind, change_addr, DUST))
+
+        tx = self.wallet.make_unsigned_transaction(coins, bch_outputs, self.config, fee, mandatory_coins=selected_slp_coins)
+        self.wallet.check_sufficient_slp_balance(slp.SlpMessage.parseSlpOutputScript(slp_op_return_msg[1]), self.config)
+        if locktime != None:
+            tx.locktime = locktime
+        if not unsigned:
+            run_hook('sign_tx', self.wallet, tx)
+            self.wallet.sign_transaction(tx, password)
```
Comment (jcramer) [inline lib/commands.py]: using wallet.sign_transaction bypasses the extra SLP validity check that we should always pre-signing any transaction type to double-check we're not burning SLP tokens.  This was added to main_window.sign_tx as shown in link below.

Just need to add this similar code:

https://github.com/simpleledger/Electron-Cash-SLP/blob/520f3ac0123ba53937db48cba697de8b265dc081/gui/qt/main_window.py#L2502

Comment (cculianu) [reply lib/commands.py]: Ah thanks. I was worried here too that maybe I'm not doing all the checks.  Will "check" this out and add this check.

Comment (cculianu) [reply lib/commands.py]: Ok, I added the check. Let me know if anything else!

**Comment by imaginaryusername** [comment]
I recall that EC-SLP also allows for "Also send BCH?" option, which can be very useful for sending some accompanying BCH in the same tx to enable receiver to starting using right away. Should there be an optional argument `amount` and become less "pure"? BCH sends go on a different output to the same address.

**Comment by cculianu** [comment]
> I recall that EC-SLP also allows for "Also send BCH?" option, which can be very useful for sending some accompanying BCH in the same tx to enable receiver to starting using right away. Should there be an optional argument `amount` and become less "pure"? BCH sends go on a different output to the same address.

Sorry, I'm not adding that to this PR. To much work already to do that.  You are free to add to it or others are.

**Comment by imaginaryusername** [comment]
That's okay, just a nice-to-have, not a must.

**Comment by cculianu** [comment]
@jcramer  -- While I have your attention -- does `commands.py` need to unconditionally call that slp_checker thing even for non-slp tx's?  It was my understanding that `wallet.make_unsigned_transaction` would not burn coins.... 

Should I add a paranoia check to the regular `_mktx` in this `Commands` class that guards against coin burning?

**Comment by cculianu** [comment]
Ok, added a change that enforces `slp_checker` magic on non-SLP tx's in `commands.py`.

**Comment by cculianu** [comment]
Note I added another PR #79 which builds on this PR and also activates the SLP engine *within* the wallet.

**Comment by cculianu** [comment]
Merged #79 , which deprecates this. Closing.

**Pull request closed without merging**